quest_test/test_workflow_metrics.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+import pytest
+
+from quest.manager import WorkflowManager
+from quest.persistence import InMemoryBlobStorage
+from quest.serializer import NoopSerializer
+
+
+@pytest.mark.asyncio
+async def test_workflow_metrics_simple():
+    storage = InMemoryBlobStorage()
+    histories = {}
+
+    def create_history(wid: str):
+        if wid not in histories:
+            histories[wid] = []
+        return histories[wid]
+
+    async def sample_workflow():
+        await asyncio.sleep(0.1)
```

Comment:
It's usually a good idea to not rely on timers for controlling the relative sequence between two tasks. Instead, use a gate.

In this situation, you can use one event to block both workflows.